src/core/PwAuth.cpp

```diff
@@ -0,0 +1,193 @@
+#include "PwAuth.hpp"
+#include "ext-session-lock-v1-protocol.h"
+#include "hyprlock.hpp"
+#include "../helpers/Log.hpp"
+#include "../config/ConfigManager.hpp"
+#include <cstddef>
+#include <cstring>
+#define GCRYPT_NO_DEPRECATED
+#define GCRYPT_NO_MPI_MACROS
+#define NEED_LIBGCRYPT_VERSION nullptr
+#include <gcrypt.h>
+
+using namespace std::chrono_literals;
+
+static auto hex2Bytes(const std::string& hex) -> std::unique_ptr<unsigned char[]> {
+    auto bytes = std::make_unique<unsigned char[]>(hex.length() / 2u);
```

Comment:
what the actual fuck? std::vector??

Reply:
I'm sorry? I just need a unique_ptr to some fixed buffer, here (for automatic deletion after the hash generation). How would you recommend doing that with `std::vector` (or, rather `std::array`)?